spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -16,55 +16,63 @@
 
+import java.nio.charset.Charset;
+import java.util.Map;
+
 import com.google.api.core.ApiFuture;
 import com.google.api.core.ApiFutureCallback;
 import com.google.api.core.ApiFutures;
 import com.google.cloud.pubsub.v1.MessageReceiver;
 import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.protobuf.ByteString;
 import com.google.pubsub.v1.PubsubMessage;
 
 import org.springframework.beans.factory.InitializingBean;
-import org.springframework.cloud.gcp.pubsub.converters.SimpleMessageConverter;
 import org.springframework.cloud.gcp.pubsub.support.PublisherFactory;
 import org.springframework.cloud.gcp.pubsub.support.SubscriberFactory;
-import org.springframework.messaging.Message;
-import org.springframework.messaging.converter.MessageConversionException;
-import org.springframework.messaging.converter.MessageConverter;
-import org.springframework.util.Assert;
 import org.springframework.util.concurrent.ListenableFuture;
 import org.springframework.util.concurrent.SettableListenableFuture;
 
 /**
+ * Default implementation of {@link PubSubOperations}.
+ *
  * @author Vinicius Carvalho
  * @author João André Martins
  */
 public class PubSubTemplate implements PubSubOperations, InitializingBean {
 
-	private MessageConverter messageConverter = new SimpleMessageConverter();
-
 	private final PublisherFactory publisherFactory;
 
 	private final SubscriberFactory subscriberFactory;
 
+	private final Charset charset;
+
 	public PubSubTemplate(PublisherFactory publisherFactory, SubscriberFactory subscriberFactory) {
+		this(publisherFactory, subscriberFactory, Charset.defaultCharset());
+	}
+
+	public PubSubTemplate(PublisherFactory publisherFactory, SubscriberFactory subscriberFactory,
+			Charset charset) {
 		this.publisherFactory = publisherFactory;
 		this.subscriberFactory = subscriberFactory;
+		this.charset = charset;
 	}
 
 	@Override
-	public ListenableFuture<String> publish(final String topic, Message message) {
-		// Convert from payload into PubsubMessage.
-		Object pubsubMessageObject =
-				this.messageConverter.fromMessage(message, PubsubMessage.class);
-
-		if (!(pubsubMessageObject instanceof PubsubMessage)) {
-			throw new MessageConversionException("The specified converter must produce "
-					+ "PubsubMessages to send to Google Cloud Pub/Sub.");
-		}
+	public ListenableFuture<String> publish(final String topic, String payload,
+			Map<String, String> headers) {
+		return publish(topic, buildPubsubMessage(payload.getBytes(this.charset), headers));
+	}
+
+	@Override
+	public ListenableFuture<String> publish(final String topic, byte[] payload,
+			Map<String, String> headers) {
+		return publish(topic, buildPubsubMessage(payload, headers));
+	}
 
-		// Send message to Google Cloud Pub/Sub.
-		ApiFuture<String> publishFuture = this.publisherFactory.getPublisher(topic)
-				.publish((PubsubMessage) pubsubMessageObject);
+	public ListenableFuture<String> publish(final String topic, PubsubMessage pubsubMessage) {
```

Comment:
Made visibility of this `public`, so users have access to the more low-level method, but didn't add it to `PubSubOperations`. LMKWYT

Reply:
I don't see a big problem to let end-users to build `PubsubMessage` by themselves and publish them as is.
So, +1 to expose this API on the interface level. See `RabbitTemplate.send(Message)`. 
Although all those non-`Message` methods are called `convertAndSend`.
We may consider the same here, however I'm fine with just overloaded variants of the `publish()`